saleor/graphql/product/mutations/products.py

```diff
@@ -469,6 +476,189 @@ class ProductCreateInput(ProductInput):
     )
 
 
+T_INPUT_MAP = List[Tuple[models.Attribute, List[str]]]
+T_INSTANCE = Union[models.Product, models.ProductVariant]
+
+
+class AttributeAssignmentMixin:
+    """Handles cleaning of the attribute input and creating the proper relations.
+
+    1. You should first call ``clean_input``, to transform and attempt to resolve
+       the provided input into actual objects.
+    2. You should then call ``is_valid`` with the cleaned input to perform a few
```

Comment:
Do we have the `is_valid` function here? If that's possible, I would try to stick to the convention that we have in model mutations, where `clean_input` is the only function you need to call to run input validation.